packages/core/src/shader/state/RenderState.ts

```diff
@@ -44,7 +45,7 @@ export class RenderState {
     const context = engine._renderContext;
     this.blendState._apply(hardwareRenderer, lastRenderState);
     this.depthState._apply(hardwareRenderer, lastRenderState);
-    this.stencilState._apply(hardwareRenderer, lastRenderState);
+    this.stencilState._apply(hardwareRenderer, lastRenderState, customStates);
```

Comment:
A little too customized.

Reply:
For all state, but now only stencil is used